src/solver/level_initializer/level_initializer_factory.h

```diff
@@ -17,11 +18,12 @@ namespace solver
         using AMRTypes = typename HybridModel::amr_types;
 
     public:
-        static std::unique_ptr<LevelInitializer<AMRTypes>> create(std::string modelName)
+        static std::unique_ptr<LevelInitializer<AMRTypes>>
+        create(std::string modelName, PHARE::initializer::PHAREDict dict)
         {
             if (modelName == "HybridModel")
             {
-                return std::make_unique<HybridLevelInitializer<HybridModel>>();
+                return std::make_unique<HybridLevelInitializer<HybridModel>>(dict);
```

Comment:
should we do std::move(dict) here?

Reply:
> should we do std::move(dict) here?

I think it's not needed as it's RVO, a copy shoud(would?) cause a compile error

Reply:
I think it's not needed as it's RVO, a copy shoud(would?) cause a compile error

Reply:
http://coliru.stacked-crooked.com/a/c5f1ed9aec4a50b4

Reply:
so it should stays as is ?

Reply:
I meant 
```
 return std::make_unique<HybridLevelInitializer<HybridModel>>(std::move(dict)); 
```

Reply:
ohh right, no I don't think it's needed
maybe just make this[1] a const ref, I added a few const functions to dict.hpp so it should be ok

[1] https://github.com/rochSmets/PHARE/blob/resistivities/src/solver/level_initializer/hybrid_level_initializer.h#L47

```
explicit HybridLevelInitializer(PHARE::initializer::PHAREDict const& dict)
```

if const fails, let me know, we should assess that I think

Reply:
done